lib/LeagueUtil.js

```diff
@@ -0,0 +1,294 @@
+const deepmerge = require('deepmerge');
+const fs = require('fs');
+const path = require('path');
+
+const ParameterError = require('./errors/ParameterError');
+
+const RateLimiter = require('./RateLimiter');
+
+class LeagueUtil {
+	// TODO: maybe split into more specific classes instead of "Util" class where everything gets dumped into
```

Comment:
@Colorfulstan That sounds fine to me, I think though that since we have a class structure for Endpoints, we could put util functions in the base class, do you think that makes sense? Either way seems fine.